Common/cpp/SharedItems/Shareable.cpp

```diff
@@ -73,6 +110,22 @@ void ShareableValue::adapt(jsi::Runtime &rt, const jsi::Value &value, ValueType
   } else {
     throw "Invalid value type";
   }
+  
+  if (value.isObject()) {
```

Comment:
have you checked this with arrays? Array data is stored under `frozenArray` and afaik isObject will return true.

The same can be a problem with host functions